default/python/AI/ShipDesignAI.py

```diff
@@ -703,30 +731,157 @@ def __init__(self):
         self.hull = None            # hull object (not hullname!)
         self.partnames = []         # list of partnames (string)
         self.parts = []             # list of actual part objects
-        self.attacks = {}           # {damage: count}
-        self.structure = 0
-        self.shields = 0
-        self.fuel = 0
-        self.speed = 0
-        self.stealth = 0
-        self.detection = 0
-        self.troops = 0
-        self.colonisation = -1      # -1 since 0 indicates an outpost (capacity = 0)
+        self._design_stats = DesignStats()
         self.production_cost = 9999
         self.production_time = 1
         self.pid = INVALID_ID               # planetID for checks on production cost if not LocationInvariant.
         self.additional_specifications = AdditionalSpecifications()
         self.design_name_dict = {k: v for k, v in zip(self.NAME_THRESHOLDS,
                                                       UserString(self.NAMETABLE, self.basename).splitlines())}
-        self.fuel_per_turn = 0
-        self.organic_growth = 0
-        self.maximum_organic_growth = 0
-        self.repair_per_turn = 0
-        self.asteroid_stealth = 0
-        self.solar_stealth = 0
-        self.fighter_capacity = 0
-        self.fighter_launch_rate = 0
-        self.fighter_damage = 0
+
+    @property
+    def attacks(self):
```

Comment:
I likewise tend to have at least a bit of a preference to just using direct access to the _design_stats object, but I also see that this approach avoids having to change a zillion references to the various stats.  I don't see an easy way to do that refactoring with PyCharm's refactoring tools, and although simple search and replace might for for most or hopefully all of it, trying to do that would add some extra uncertainty/risk and need for more testing.  

I think it is probably best to go with this PR's approach for now, and we could consider refactoring more later on.